src/clients/python/src/tigerbeetle_python/bindings.py

```diff
@@ -0,0 +1,733 @@
+            debits_posted=self.debits_posted.to_python(),
+            credits_pending=self.credits_pending.to_python(),
+            credits_posted=self.credits_posted.to_python(),
+            timestamp=self.timestamp,
+        )
+
+CAccountBalance._fields_ = [ # noqa: SLF001
+    ("debits_pending", c_uint128),
+    ("debits_posted", c_uint128),
+    ("credits_pending", c_uint128),
+    ("credits_posted", c_uint128),
+    ("timestamp", ctypes.c_uint64),
+    ("reserved", ctypes.c_uint8 * 56),
+]
+
+
+class CQueryFilter(ctypes.Structure):
+    @classmethod
+    def from_param(cls, obj):
+        if obj.user_data_64 > 2**64 - 1:
+            raise IntegerOverflowError("user_data_64 is too large to fit \
+in 64 bits (" + str(obj.user_data_64) + ")")
+        if obj.user_data_32 > 2**32 - 1:
+            raise IntegerOverflowError("user_data_32 is too large to fit \
+in 32 bits (" + str(obj.user_data_32) + ")")
+        if obj.ledger > 2**32 - 1:
+            raise IntegerOverflowError("ledger is too large to fit \
+in 32 bits (" + str(obj.ledger) + ")")
+        if obj.code > 2**16 - 1:
+            raise IntegerOverflowError("code is too large to fit \
+in 16 bits (" + str(obj.code) + ")")
+        if obj.timestamp_min > 2**64 - 1:
+            raise IntegerOverflowError("timestamp_min is too large to fit \
+in 64 bits (" + str(obj.timestamp_min) + ")")
+        if obj.timestamp_max > 2**64 - 1:
+            raise IntegerOverflowError("timestamp_max is too large to fit \
+in 64 bits (" + str(obj.timestamp_max) + ")")
+        if obj.limit > 2**32 - 1:
+            raise IntegerOverflowError("limit is too large to fit \
+in 32 bits (" + str(obj.limit) + ")")
+        return cls(
+            user_data_128=c_uint128.from_param(obj.user_data_128),
+            user_data_64=obj.user_data_64,
+            user_data_32=obj.user_data_32,
+            ledger=obj.ledger,
+            code=obj.code,
+            timestamp_min=obj.timestamp_min,
+            timestamp_max=obj.timestamp_max,
+            limit=obj.limit,
+            flags=obj.flags,
+        )
+
+
+    def to_python(self):
+        return QueryFilter(
+            user_data_128=self.user_data_128.to_python(),
+            user_data_64=self.user_data_64,
+            user_data_32=self.user_data_32,
+            ledger=self.ledger,
+            code=self.code,
+            timestamp_min=self.timestamp_min,
+            timestamp_max=self.timestamp_max,
+            limit=self.limit,
+            flags=QueryFilterFlags(self.flags),
+        )
+
+CQueryFilter._fields_ = [ # noqa: SLF001
+    ("user_data_128", c_uint128),
+    ("user_data_64", ctypes.c_uint64),
+    ("user_data_32", ctypes.c_uint32),
+    ("ledger", ctypes.c_uint32),
+    ("code", ctypes.c_uint16),
+    ("reserved", ctypes.c_uint8 * 6),
+    ("timestamp_min", ctypes.c_uint64),
+    ("timestamp_max", ctypes.c_uint64),
+    ("limit", ctypes.c_uint32),
+    ("flags", ctypes.c_uint32),
+]
+
+
+@dataclasses.dataclass
```

Comment:
Let's maybe use a frozen instance here?

https://docs.python.org/3/library/dataclasses.html#frozen-instances

Reply:
And I _think_ we might also want to pass `slots=true`? There's going to be a lot of these, so might as well optimize memory usage!

Reply:
> Let's maybe use a frozen instance here?

Hmm, I'm torn on this - I feel like it might make things harder, especially things like building the object as you go which might have a valid use case.

> And I think we might also want to pass slots=true? There's going to be a lot of these, so might as well optimize memory usage!

Yeah I think it's worth it - `slots` was only added in 3.10, but we can add it if supported.